include/triton/Dialect/TritonGPU/IR/TritonGPUOps.td

```diff
@@ -394,6 +393,28 @@ def TTG_RemoteShmemStoreOp : TTG_Op<"remote_shmem_store"> {
   }];
 }
 
+def TTG_AsyncRemoteShmemStoreOp : TTG_Op<"async_remote_shmem_store"> {
```

Comment:
Please add a comment to this and RemoteShmemStoreOp to indicate that [MemWrite<SharedMemory> is not set on purpose because we prefer manual synchronization for now.

Reply:
Actually this is a good point. The flag should be set to reflect side effects?

Reply:
Yeah. In the case of LocalStoreOp MembarAnalysis uses this for inserting correct bar.syncs() for local shmem writes. We don't set this here to prevent MembarAnalysis from considering remote stores.

Reply:
I see. But we need something here to prevent compiler code motioning the remote store over the barrier wait. Also for async remote store, we need to prevent it being folded with another store.